app/helpers/application_helper.rb

```diff
@@ -40,9 +40,10 @@ def number_to_money(number)
     number_to_currency(number, unit: "£", separator: ".", delimiter: ",", precision: 2)
   end
 
-  def variable_controller_benefit_path(action: nil, id: nil)
+  def variable_controller_benefit_path(action: nil, id: nil, partner: false)
     prefix = "#{action}_" if action
-    controller = controller_name == "check_answers" ? "check_benefits_answer" : "benefit"
+    main_name = partner ? "partner_benefit" : "benefit"
+    controller = controller_name == "check_answers" ? "check_#{main_name}s_answer" : main_name
 
     if id
```

Comment:
This now looks *really* complicated. Can we look to refactor it? WDYT?

Reply:
Moved into benefits_helper and split up into some simpler parts.